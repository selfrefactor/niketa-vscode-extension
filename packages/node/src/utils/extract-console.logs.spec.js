import { extractConsoleLogs } from './extract-console.logs'

const testInput =
  'console.log src/_modules/parseBeforeNotify.spec.js:10\n' +
  '    string\n' +
  '\n' +
  '  console.log src/_modules/parseBeforeNotify.spec.js:12\n' +
  '    string [ 1, 23, { a: 1 } ]\n' +
  '\n'

const testInputSecond = `
console.log source/adjust.js:6
1 [ 0, 1, 2 ]

console.log source/adjust.js:6
1 [ 0, 1, 2 ]

console.log source/adjust.js:6
1 [ 0, 1, 2 ]

console.log source/adjust.js:6
1 [ 0, 1, 2 ]

console.log source/adjust.js:6
-2 [ 0, 1, 2 ]

console.log source/adjust.js:6
4 [ 0, 1, 2, 3 ]

console.log source/adjust.js:6
-5 [ 0, 1, 2, 3 ]

-----------|---------|----------|---------|---------|-------------------
File       | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
-----------|---------|----------|---------|---------|-------------------
All files  |     100 |      100 |     100 |     100 |                   
adjust.js |     100 |      100 |     100 |     100 |                   
-----------|---------|----------|---------|---------|-------------------
`.trim()

test('happy', () => {
  const result = extractConsoleLogs(testInput)
  expect(result).toMatchSnapshot()
})

test.only('multiple logs on same line', () => {
  const result = extractConsoleLogs(testInputSecond)
  expect(result).toMatchSnapshot()
})
