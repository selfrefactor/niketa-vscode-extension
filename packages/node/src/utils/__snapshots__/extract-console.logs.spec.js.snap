// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happy 1`] = `
Object {
  "src/_modules/parseBeforeNotify.spec.js": Array [
    Object {
      "decoration": "string",
      "line": "10",
    },
    Object {
      "decoration": "string [ 1, 23, { a: 1 } ]",
      "line": "12",
    },
  ],
}
`;

exports[`multiple logs on same line 1`] = `
Object {
  "source/adjust.js": Array [
    Object {
      "decoration": "1 [ 0, 1, 2 ], 1 [ 0, 1, 2 ], 1 [ 0, 1, 2 ], 1 [ 0, 1, 2 ], -2 [ 0, 1, 2 ], 4 [ 0, 1, 2, 3 ], -5 [ 0, 1, 2, 3 ]",
      "line": "6",
    },
  ],
}
`;
